File: TheLeague/ClientApp/app/models/transfer.ts
export class Transfer {
    id: number;
    transferDate: string;
    teamTransfers: TeamTransfer[];
}

export class TeamTransfer {
    transferredTo: number;
    playersTransferred: PlayerTransferred[];
}

export class PlayerTransferred {
    transferredFrom: number;
    playerId: number;
}

export class PlayerTransferredItem {
    transferredFromTeamName: string;
    transferredToTeamName: string;
    playerName: string;
}

export class FullTransferItem {
    transferDate: string;
    playersTransferred: PlayerTransferredItem[];
}

File: TheLeague/SharedModels/Transfer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson.Serialization.Attributes;

namespace TheLeague.SharedModels {
    public class Transfer {
        [BsonId]
        public Guid Id;

        public int Season;
        public List<TeamTransfer> TeamTransfers;

        public DateTime TransferDate;

        public Transfer(List<TeamTransfer> teamTransfers, DateTime transferDate, int season) {
            Id = Guid.NewGuid();
            TeamTransfers = teamTransfers;
            TransferDate = transferDate;
            Season = season;
        }
    }

    public class TeamTransfer {
        public int TransferredTo;
        public List<PlayersTransferred> PlayersTransferred;

        public TeamTransfer(int transferredTo, List<PlayersTransferred> playersTransferred) {
            TransferredTo = transferredTo;
            PlayersTransferred = playersTransferred;
        }
    }

    public class PlayersTransferred {
        public int PlayerId;
        public int TransferredFrom;

        public PlayersTransferred(int playerId, int trasferredFrom) {
            PlayerId = playerId;
            TransferredFrom = trasferredFrom;
        }
    }
}
